inc/tokens/class-authorization-code.php

```diff
@@ -108,6 +108,36 @@ public function get_expiration() {
 		return (int) $value['expiration'];
 	}
 
+	private function validate_code_verifier( $args ) {
+		$value = $this->get_value();
+
+		if ( empty( $value['code_challenge'] ) ) {
+			return true;
+		}
+
+		$code_verifier = $args['code_verifier'];
+		$is_valid = false;
+
+		switch ( strtolower( $value['code_challenge_method'] ) ) {
+			case 's256':
+				$decoded = base64_encode( hash( 'sha256', $code_verifier ) );
+				$is_valid = $decoded === $value['code_challenge'];
+				break;
+			case 'plain':
+				$is_valid = $code_verifier === $value['code_challenge'];
```

Comment:
Both this equality check and the one above should use `hash_equals()` to ensure constant-time string comparison (to avoid timing attacks).